src/projections/bonne.cpp

```diff
@@ -58,32 +58,43 @@ static PJ_LP bonne_s_inverse(PJ_XY xy, PJ *P) { /* Spheroidal, inverse */
     double rh;
 
     xy.y = Q->cphi1 - xy.y;
-    rh = hypot(xy.x, xy.y);
+    rh = copysign(hypot(xy.x, xy.y), Q->phi1);
```

Comment:
minor nit is to have some consistency with L86 (and remove L58)
```suggestion
    const double rh = copysign(hypot(xy.x, xy.y), Q->phi1);
```